djangosaml2/tests/__init__.py

```diff
@@ -141,7 +144,10 @@ def test_login_one_idp(self):
         self.assertIn('RelayState', params)
 
         saml_request = params['SAMLRequest'][0]
-        expected_request = """<samlp:AuthnRequest xmlns:saml="urn:oasis:names:tc:SAML:2.0:assertion" xmlns:samlp="urn:oasis:names:tc:SAML:2.0:protocol" AssertionConsumerServiceURL="http://sp.example.com/saml2/acs/" Destination="https://idp.example.com/simplesaml/saml2/idp/SSOService.php" ID="XXXXXXXXXXXXXXXXXXXXXX" IssueInstant="2010-01-01T00:00:00Z" ProtocolBinding="urn:oasis:names:tc:SAML:2.0:bindings:HTTP-POST" Version="2.0"><saml:Issuer Format="urn:oasis:names:tc:SAML:2.0:nameid-format:entity">http://sp.example.com/saml2/metadata/</saml:Issuer><samlp:NameIDPolicy AllowCreate="false" Format="urn:oasis:names:tc:SAML:2.0:nameid-format:persistent" /></samlp:AuthnRequest>"""
+        if PY_VERSION < (3, 8):
```

Comment:
Why is this needed now?

Reply:
The ordering of the attributes in the xml is different on py3.8. I don't know why, but this is something out of our control. Due to this, the hardcoded string comparison fails (while they represent equal xml documents)